src/ui/inventory.js

```diff
@@ -0,0 +1,182 @@
+            }
+            // Add items into first empty slot
+            else if (index >= 0) {
+                this.playerItems[index] = {
+                    name: item.item, 
+                    material: item.material
+                };
+            }
+            // Add new item to end
+            else if (this.playerItems.length == this.inventory.length) {
+                this.playerItems.push({
+                    name: item.item, 
+                    material: item.material
+                }); 
+                index = this.inventory.length;
+            }
+
+            // Add item images
+            let column = index % 4;
+            let row = Math.floor(index / 4);
+            let x = 570 + column * 45;
+            let y = 225 + row * 35;
+            item.addToInventory(x, y, index);
+
+            // Hide if inventory is not selected
+            if (this.scene.currentPanel != CONSTANTS.PANEL.INVENTORY) {
+                item.show(false);
+            }
+
+            this.inventory[index] = item;
+        } else if (this.inventory.length < 28) {
+            this.inventory.push({});
+        } else {
+            console.log("Inventory is full");
+        }
+    }
+
+    async checkForEmptySlot() {
+        for (let index = 0; index < this.playerItems.length; index++) {
+            if (!Object.keys(this.playerItems[index]).length) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    createRightClickMenu(x, y, item, index) {
+        let menuBox = this.scene.add
+            .image(x, y, "right-click-menu")
+            .setDepth(4)
+            .setInteractive()
+            .on("pointerout", pointer => {
+                // Check to ensure it doesn't trigger when hovering over text options
+                if (
+                    pointer.worldX > x + menuBox.width / 2 ||
+                    pointer.worldY > y + menuBox.height / 2 ||
+                    pointer.worldX < x - menuBox.width / 2 ||
+                    pointer.worldY < y - menuBox.height / 2
+                ) {
+                    this.menu.destroy();
+                }
+            });
+
+        // Add text options
+        // Have to use two separate texts per option for different clors
+        let options = [menuBox];
+        let optionsY = y -45;
+
+        // Generate dynamic list of actions (use, wield, bury, etc.)
+        item.actions.forEach(action => {
+            optionsY += 15;
+            let itemText = this.scene.add.text(x + 10, optionsY, item.name, FONTS.ITEM_NAME);
+            let actionText = this.scene.add.text(x - 48, optionsY, action, FONTS.OPTIONS_MENU)
+                .setInteractive()
+                .setDepth(5)
+                .on("pointerdown", () => {
+                    item[action]();
+                    this.menu.destroy();
+                });
+
+            options.push(actionText);
+            options.push(itemText);
+        });
```

Comment:
You might notice in the right click menu that the capitalization is not consistent (use item vs Examine item). This is because when I dynamically create the text and action I use this syntax: `item[action]();` which is essentially item.use(); or item.bury();, etc. 

The simplest way to fix this would to change "use" to "Use" in both the item.actions[] array as well as the function name itself. Function names being capitalized is against convention however so I was hesitant to do this. lmk what you think

Reply:
We will definitely want them to be capitalized. It would probably make more sense for a lot of the specific option stuff to be within an item class. I.e. the items would have a method that returns all of its options and this will just generate the interactive image.

Reply:
If i'm understanding correctly that's what this already does. 
The ItemTypes class defines the methods "drop" and "examine" because they exist for all items. Then each respective subclass defines an array of additional options such as "use", "bury", "wield", etc. Thinking about it now though, I think all items have a "use" options so I can probably move that.

Other than capitalization is there anything you would change for this?

Reply:
See if you can get it in a single loop:
for optionText, optionMethod in item.options -> item.options returns hashmap of text : -> method
    generate one line of background image
    add text and set interactive with the method

Reply:
This could be done by moving the use, drop, and examine methods into ItemType.actions[]. Then in the child classes instead of creating the actions array you just push new actions to front.

Reply:
Also for the capitalization, I think I would do this instead of what I have now:
```
actions = [{text: "Use", func: "use"}, 
           {text: "Drop", func: "drop"}, 
           {text: "Examine", func: "examine"}];
...
let actionText = this.scene.add.text(x - 48, optionsY, action.text, FONTS.OPTIONS_MENU)
item[action.func]();
```

Adds just a little bit more into the ItemType class but allows for flexibility

Reply:
By ItemType class do you mean Item class? They should have a default implementation in the item class if we think all items will have those options (they should, although some items in RS have destroy instead of drop to prevent duplicates by re-picking up). That looks good for the capitalization.

Reply:
Yeah right now I have it called ItemType I think because there was already an Item class defined for use in the shop. I think i'll rename ItemType to Item when I make these changes but keep the other Item class for now. The shop stuff can be modified in a later PR.